x-pack/plugins/encrypted_saved_objects/server/crypto/encrypted_saved_object_type_definition.ts

```diff
@@ -9,15 +9,32 @@ import type { EncryptedSavedObjectTypeRegistration } from './encrypted_saved_obj
 
 /**
  * Represents the definition of the attributes of the specific saved object that are supposed to be
- * encrypted. The definition also dictates which attributes should be excluded from AAD and/or
+ * encrypted. The definition also dictates which attributes should be included in AAD and/or
  * stripped from response.
  */
 export class EncryptedSavedObjectAttributesDefinition {
   public readonly attributesToEncrypt: ReadonlySet<string>;
-  private readonly attributesToExcludeFromAAD: ReadonlySet<string> | undefined;
+  private readonly attributesToIncludeInAAD: ReadonlySet<string> | undefined;
   private readonly attributesToStrip: ReadonlySet<string>;
 
   constructor(typeRegistration: EncryptedSavedObjectTypeRegistration) {
+    if (typeRegistration.attributesToIncludeInAAD) {
+      const invalidAttributeKeys = new Array<string>();
+      typeRegistration.attributesToEncrypt.forEach((attribute) => {
+        const attributeKey = typeof attribute !== 'string' ? attribute.key : attribute;
+        if (typeRegistration.attributesToIncludeInAAD?.has(attributeKey)) {
```

Comment:
__question:__ 

`typeRegistration.attributesToIncludeInAAD-->?<--.has(attributeKey)`

does TS still complain about `typeRegistration.attributesToIncludeInAAD` being potentially `undefined` even though we're within `if (typeRegistration.attributesToIncludeInAAD) {` block?

Reply:
It does 😞